test/server/index.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2015 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+'use strict';
+
+// This server is needed most importantly to add the Service-Worker-Allowed
+// header to any service worker loaded in the tests. This allows the scope
+// to be manipulated any way we want / need during testing.
+
+var path = require('path');
+var express = require('express');
+var app = express();
+
+// Set up static assets
+app.use('/test/browser-tests/',
+  express.static(path.join(__dirname, '../browser-tests/'), {
```

Comment:
Aaahhhh clever. Got it. Thank you. I got all those things fixed anyway.

On Wed, 13 Jan 2016, 20:11 Jeffrey Posnick <notifications@github.com> wrote:

> In test/server/index.js
> <https://github.com/GoogleChrome/sw-toolbox/pull/70#discussion_r49643887>:
>
> > + *
> > + */
> > +
> > +'use strict';
> > +
> > +// This server is needed most importantly to add the Service-Worker-Allowed
> > +// header to any service worker loaded in the tests. This allows the scope
> > +// to be manipulated any way we want / need during testing.
> > +
> > +var path = require('path');
> > +var express = require('express');
> > +var app = express();
> > +
> > +// Set up static assets
> > +app.use('/test/browser-tests/',
> > +  express.static(path.join(__dirname, '../browser-tests/'), {
>
> What I meant was that path.join() will construct paths using path.sep as
> the delimiter, which is normally what you'd want
> <https://gist.github.com/domenic/2790533#paths-and-urls> to ensure that
> your code works on both Windows and UNIX-y platforms.
>
> But if you pass in some strings that have / hardcoded in them already,
> you lose that benefit, since you'll end up with a string containing a mix
> of \ and / on Windows. That's why I'm suggesting that you pass in each
> path component as a separate argument.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/sw-toolbox/pull/70/files#r49643887>.
>